common/util.go

```diff
@@ -525,3 +525,12 @@ func GetDefaultAdvancedVisibilityWritingMode(isAdvancedVisConfigExist bool) stri
 	}
 	return AdvancedVisibilityWritingModeOff
 }
+
+// GetDomainTagByName returns the corresponding tag if the domain name could be found, otherwise return un-known tag
+func GetDomainTagByName(domainName string, err error) metrics.Tag {
```

Comment:
This function signature looks kind of weird to me. How about `GetDomainTagByName(domainName string) metrics.Tag` and if domainName is empty return the unknown tag.

Actually I would suggest move the entire function to history package to avoid the cycle dependency issue. Otherwise, you still need to have an if/else statement every place you call the updated function.